pkg/controllers/edgeconnect/controller.go

```diff
@@ -676,32 +741,100 @@ func (controller *Controller) createOrUpdateEdgeConnectDeployment(ctx context.Co
 		return err
 	}
 
+	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
+	if err != nil {
+		_log.Debug("building EdgeConnect client failed")
+
+		return err
+	}
+
+	err = controller.createOrUpdateConnectionSetting(ctx, edgeConnectClient, edgeConnect, edgeConnectToken)
+	if err != nil {
+		_log.Debug("creating EdgeConnect connection setting failed")
+
+		return err
+	}
+
 	_log.Debug("EdgeConnect deployment created/updated successfully")
 
 	return nil
 }
 
-func (controller *Controller) createOrUpdateEdgeConnectConfigSecret(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) (string, error) {
+func (controller *Controller) createOrUpdateConnectionSetting(ctx context.Context, edgeConnectClient edgeconnect.Client, edgeConnect *edgeconnectv1alpha1.EdgeConnect, latestToken string) error {
+	_log := log.WithValues("namespace", edgeConnect.Namespace, "name", edgeConnect.Name)
+
+	var kubeSystemNamespace corev1.Namespace
+
+	err := controller.apiReader.Get(ctx, client.ObjectKey{Name: kubeSystemNamespaceName}, &kubeSystemNamespace)
+	if err != nil {
+		return err
+	}
+
+	envSetting, err := edgeConnectClient.GetConnectionSetting(string(kubeSystemNamespace.UID))
+	if err != nil {
+		_log.Info("Failed getting EdgeConnect connection setting object")
+
+		return err
+	}
+
+	if (envSetting == edgeconnect.EnvironmentSetting{}) {
+		_log.Debug("Creating edgeconnect connection setting object...")
+
+		err = edgeConnectClient.CreateConnectionSetting(
+			edgeconnect.EnvironmentSetting{
+				SchemaId:      edgeconnect.KubernetesConnectionSchemaID,
+				SchemaVersion: edgeconnect.KubernetesConnectionVersion,
+				Scope:         edgeconnect.KubernetesConnectionScope,
+				Value: edgeconnect.EnvironmentSettingValue{
+					Name:      edgeConnect.Name,
+					Uid:       string(kubeSystemNamespace.UID),
+					Namespace: edgeConnect.Namespace,
+					Token:     latestToken,
+				},
+			},
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	if (envSetting != edgeconnect.EnvironmentSetting{}) {
+		if envSetting.Value.Token != latestToken {
+			_log.Debug("Updating EdgeConnect connection setting object...")
+
+			envSetting.Value.Token = latestToken
+			err = edgeConnectClient.UpdateConnectionSetting(envSetting)
+		}
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (controller *Controller) createOrUpdateEdgeConnectConfigSecret(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) (string, string, error) {
```

Comment:
if you are going to have multiple return values (ie. not 1 value and 1 error), AND each value does not have a distinct type, then I would recommend naming them, so its easier to understand which value is supposed to be what. (otherwise you have to sniff through the code of the function to figure it out)

```suggestion
func (controller *Controller) createOrUpdateEdgeConnectConfigSecret(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) (token string, hash string, err error) {
```